src/Middleware/WebSockets/src/WebSocketMiddleware.cs

```diff
@@ -6,6 +6,7 @@
 using Microsoft.AspNetCore.Builder;
 using Microsoft.AspNetCore.Http;
 using Microsoft.AspNetCore.Http.Features;
+using Microsoft.AspNetCore.Http.Timeouts;
```

Comment:
@halter73 do we really think this namespace is better than reusing the features namespace?

Reply:
I think it's fine. I suspect the WebSocket middleware is unique in that it uses other standard features not tied to middleware like the `IHttpUpgradeFeature` so it's already using `Microsoft.AspNetCore.Http.Features`.

I suspect most consumers of `IHttpRequestTimeoutFeature` won't be using other standard request features and will expect to find it in `Microsoft.AspNetCore.Http.Timeouts` as most of the other public types related to the feature including the `RequestTimeoutAttribute` and `RequestTimeoutPolicy`. This is the way we did it for output caching and exception handler too.

The WebSocket middleware and the `IHttpWebSocketFeature` is the big counter example of course. You could argue that `IEndpointFeature` and `IRouteValuesFeature` are set by endpoint routing middleware, but that's much more core. I think that covers all the middleware we ship that do add interfaces in the `Microsoft.AspNetCore.Http.Features` namespace.

I don't feel super strongly, but request timeout feels more like output caching and less core than WebSockets or routing.

Reply:
I feel like this doesn’t improve the current situation and adds to the namespace blazor that we tried to fix with global usings. A new namespace should be made if there’s going to be a significant amount of types introduced otherwise it feels like we’ve done a poor job. We should not introduce a new features sub namespace until there’s a good reason to do so.